collab2/foraging/toolkit/proximity.py

```diff
@@ -0,0 +1,93 @@
+import math
+
+import numpy as np
+import pandas as pd
+
+
+def proximity_score(distance, getting_worse=1.5, optimal=4, proximity_decay=1):
+    if distance <= getting_worse:
+        return math.sin(math.pi / (2 * getting_worse) * (distance + 3 * getting_worse))
+    elif distance <= getting_worse + 1.5 * (optimal - getting_worse):
+        return math.sin(
+            math.pi / (2 * (optimal - getting_worse)) * (distance - getting_worse)
+        )
+    else:
+        return math.sin(
+            math.pi
+            / (2 * (optimal - getting_worse))
+            * (1.5 * (optimal - getting_worse))
+        ) * math.exp(1) ** (
+            -proximity_decay * (distance - optimal - 0.5 * (optimal - getting_worse))
+        )
+
+
+def foragers_to_forager_distances(foragers):
+    forager_distances = []
```

Comment:
I remember we discussed this, but I don't remember what we agreed to: did we decide to leave "forager" rather than "agent"? I remember you complaining that your fish aren't foraging. :)

Reply:
We decided to use foragers for now, and have a different PR renaming everything once we are done!